lib/publiccloud/utils.pm

```diff
@@ -460,20 +462,19 @@ Transactional systems like SLE micro used C<transactional_update up> and reboot.
 
 sub ssh_update_transactional_system {
     my ($instance) = @_;
-    my $cmd_time = time();
-    my $cmd = "sudo transactional-update -n up";
     my $cmd_name = "transactional update";
-    # first run, possible update of packager
-    my $ret = $instance->ssh_script_run(cmd => $cmd, timeout => 1500);
-    $instance->softreboot(timeout => get_var('PUBLIC_CLOUD_REBOOT_TIMEOUT', 600));
-    record_info($cmd_name, 'The command ' . $cmd_name . ' took ' . (time() - $cmd_time) . ' seconds.');
-    die "$cmd_name failed with $ret" if ($ret != 0 && $ret != 102 && $ret != 103);
-    # second run, full system update
-    $cmd_time = time();
-    $ret = $instance->ssh_script_run(cmd => $cmd, timeout => 6000);
-    $instance->softreboot(timeout => get_var('PUBLIC_CLOUD_REBOOT_TIMEOUT', 600));
-    record_info($cmd_name, 'The second command ' . $cmd_name . ' took ' . (time() - $cmd_time) . ' seconds.');
-    die "$cmd_name failed with $ret" if ($ret != 0 && $ret != 102);
+    my @t = (1500, 6000);
+    my $code = 103;
+    # 1st run, possible update of packager
+    # 2nd run, full system update
+    for (0 .. 1) {
+        my $cmd_time = time();
+        # run "sudo transactional-update -n up";
+        my $ret = $instance->zypper_call_remote(timeout => $t[$_]);
```

Comment:
this does not look nice . we should not expect from reader to know what is default . better to explicitly write `-n up`

Reply:
Also the `subcommand` has by **default** `up / update`, applied on empty  command input.

Anyway, still one can opt to use the _full-command_ form and provide the complete command to execute, skipping any default or pre-elaboration.